docs/compatibility.md

```diff
@@ -15,6 +15,14 @@ problem. Please also look at the current list of
 [bugs](https://github.com/NVIDIA/spark-rapids/issues?q=is%3Aopen+is%3Aissue+label%3Abug) which are
 typically incompatibilities that we have not yet addressed.
 
+## Non-empty nulls
+
+The SQL plugin doesn't support nested types to have non-empty nulls and if data being processed 
```

Comment:
So this is all for the tests and I am not sure it belongs in compatibility.md

Production use cases won't see this assertion. 

Would `CONTRIBUTING.md` be a better fit for this?

Reply:
This isn't internal. Anyone passing data through our plugin that contains non-empty nulls will see this `AssertionError` unless they turn off assertions using the above mentioned Java command line parameter

Reply:
How can the user control this???? Assertions are also off by default in java.  You have to enable them with `-ea` or something similar. This is not a user facing problem. 

If they pass in arrow data and we get an error because of this, then we have a BUG in our arrow translation.  If they use the java APIs directly to write UDFs, then I can see it, but it should be a part of the docs for the CUDF java APIs. Not here, and especially not as the first thing in the compatibility docs.

Reply:
So we are asking two questions here. 
1. Is adding this documentation needed at all? 

I think it does until we have fixed the `mergeAndSetValidity` method and everywhere else where we can be introducing non-empty nulls. I wasn't sure if my [fix](https://github.com/rapidsai/cudf/pull/13335) for `mergeAndSetValidity` will make it in before 23.06 is released and until it does I imagine a user who has enabled assertions and consuming arrow data will see this.

2. Where does this belong?

I wasn't sure where to put it, I think it belongs in compatibility because it has something to do with consuming external data which can be considered valid by some 3rd party tools like Arrow. We can put it wherever we have a consensus.

Reply:
The assertions are off by default. The only time that a user would see this is if they are running unit tests against our code or enable assertions themselves for some reason and happen to run into one of these problems.


I think this needs to be in an FAQ and not compatibility. This is not a difference between Apache Spark with and without our plugin. Turning on our plugin will still produce the same results as before.

Reply:
> I think this needs to be in an FAQ 

I would argue this doesn't meet the "frequently" part of FAQ.  Agree with Bobby that the plugin will do the same thing as Apache Spark here.

What, specifically, are we worried about here that users really need to know about?  What is the use case where users would encounter it?  If there are no use cases, then I don't see why we need to put this in user-facing documentation.